core/src/global.rs

```diff
@@ -179,19 +193,43 @@ pub fn get_chain_type() -> ChainTypes {
 	})
 }
 
-/// One time initialization of the global chain_type.
+/// One time initialization of the global future time limit
 /// Will panic if we attempt to re-initialize this (via OneTime).
-pub fn init_global_chain_type(new_type: ChainTypes) {
-	GLOBAL_CHAIN_TYPE.init(new_type)
+pub fn init_global_future_time_limit(new_ftl: u64) {
+	GLOBAL_FUTURE_TIME_LIMIT.init(new_ftl)
 }
 
-/// One time initialization of the global chain_type.
+/// Set the future time limit on a per-thread basis via thread_local storage.
+pub fn set_local_future_time_limit(new_ftl: u64) {
+	FUTURE_TIME_LIMIT.with(|ftl| ftl.set(Some(new_ftl)))
+}
+
+/// Future Time Limit (FTL)
+/// Look at thread local config first. If not set fallback to global config.
+/// Default to false if global config unset.
+pub fn get_future_time_limit() -> u64 {
+	FUTURE_TIME_LIMIT.with(|ftl| match ftl.get() {
+		None => {
+			if GLOBAL_FUTURE_TIME_LIMIT.is_init() {
+				let ftl = GLOBAL_FUTURE_TIME_LIMIT.borrow();
+				set_local_future_time_limit(ftl);
+				ftl
+			} else {
+				// Global config unset, default to 5 minutes
+				5 * 60
```

Comment:
We should call `set_local_future_time_limit()` either way here so we just read local config next time.
This way we are simply lazily initializing the local config.

Reply:
Does that apply to is_nrd_enabled() as well?

Reply:
set_local_future_time_limit() always called now